### DISCUSSION TITLE
Usage examples for tmp_path_factory

### DISCUSSION THREAD

**Discussion by pawamoy** [pytest-dev/pytest#8602] [General] [answered]
I had trouble finding how I'm supposed to use the `tmp_path_factory` fixture. I had to:

- see that it returns an instance of `TempPathFactory`, and go read the source of this class, then see it has a `mktemp` method to create a directory
- understand it replaces `tmpdir_factory`, guess it has the same usage and read the docs for `tmpdir_factory` instead

It's still not clear if `tmpdir_factory` is deprecated or not in favor of `tmp_path_factory`. If it is, maybe examples and docs should be moved from `tmpdir_factory` to `tmp_path_factory`, and the former would link to the latter, instead of the other way around.

What do you think users/maintainers?

**Comment by bluetech** [comment, marked as answer]
`TempPathFactory`/`tmp_path_factory` is preferred over `tmpdir_factory` -- we are transitioning from `py.path.local` to `pathlib.Path`. It is still not officially deprecated but will be soon.

Which documentation page (URL) did you look at?

**Reply by pawamoy** [reply]
OK, thanks 🙂 

> Which documentation page (URL) did you look at?

Everything with `tmp_path_factory` in it 😅 

- https://docs.pytest.org/en/stable/reference.html?highlight=tmp_path_factory#std-fixture-_pytest.tmpdir.tmp_path_factory
- https://docs.pytest.org/en/stable/reference.html?highlight=tmp_path_factory
- https://docs.pytest.org/en/stable/tmpdir.html?highlight=tmp_path_factory
- https://docs.pytest.org/en/stable/example/simple.html?highlight=tmp_path_factory
- https://docs.pytest.org/en/stable/contents.html?highlight=tmp_path_factory
- https://docs.pytest.org/en/stable/builtin.html?highlight=tmp_path_factory
- https://docs.pytest.org/en/stable/fixture.html?highlight=tmp_path_factory

**Reply by bluetech** [reply]
Can you try the `latest` docs https://docs.pytest.org/en/latest/ and see if the situation is improved there? We've had some improvements there lately. This will become the `stable` once pytest 6.3 is released.

**Reply by pawamoy** [reply]
Ah, yes, it's perfect! I immediately found what I was looking for 🙂 
Sorry for not thinking about checking the latests docs!
